addons/lunch/static/src/js/lunch.js

```diff
@@ -0,0 +1,49 @@
+(function() {
+
+    "use strict";
+
+    var QWeb = openerp.web.qweb,
+        lunch = openerp.lunch = {};
+
+    lunch.PreviousOrder = openerp.web.form.AbstractField.extend({
+        template: 'lunch.PreviousOrder',
+        set_value: function (value_) {
+            var values = JSON.parse(value_);
+            this._super(values);
+        },
+        update_lines: function (pref_id, records) {
+            var self = this;
+            var value = this.get_value();
+            var categories = value.categories;
+            _.each(categories, function(category) {
+                _.each(category, function(prod_obj, prod_id) {
+                    _.each(prod_obj, function(prod_detail, prod_note) {
+                        if (prod_detail.pref_id == parseInt(pref_id)) {
+                            if(prod_note == 'False') { prod_note = ''; }
+                            records.push([0, 0, {'product_id': parseInt(prod_id), 'note': prod_note, 'price': prod_detail.price}]);
+                            self.field_manager.set_values({order_line_ids: records});
+                        }
+                    });
+                });
+            });
```

Comment:
No need to apply this type of logic here, just store and bind  value with DOM element and on click need to add.